### ISSUE TITLE
PresentationReconciler does not handle multi line rules correctly

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform.ui#1330]
PDE has an extension of `PresentationReconciler` (`TargetPlatformPresentationReconciler`) that mentions broken MultiLine Rules.

To reproduce the bug do the following:

1. In `org.eclipse.pde.internal.genericeditor.target.extension.reconciler.presentation.TargetPlatformPresentationReconciler` uncomment the override method `createPresentation`
2. open a target file and add a multi-line comment with `<!--   -->`
3. This results in that part become green
4. Now edit inside the multiline comment (e.g. add a space)
5. Now parts become colored again

https://github.com/eclipse-platform/eclipse.platform.ui/assets/1331477/8d3c753d-7b43-4607-9d8f-257036732dc6

Instead of having some hacks, this should be fixed in the source, especially as the PDE implementation is flawed and does not support different document partitions what is important for extenders like m2e. 

@mickaelistria can you help her or know someone that is familiar with the damage/repair framework?

**Comment by mickaelistria** [comment]
> can you help her or know someone that is familiar with the damage/repair framework?

Sorry, I'm busy with other higher priority things; and don't know anyone that could magically fix it better and faster than any of the currently active maintainers on Platform.

**Comment by laeubi** [comment]
I don't think it requires magic, but a good understanding of how document reconciles are supposed to work I couldn't find anything to work it out for myself and the "fix" of PDE to simply reconcile everything seem to defeats the purpose of it.

So I hope that at least this quite direct case to reproduce the issue might help someone with deeper knowledge to take a look and maybe has an idea.

**Comment by szarnekow** [comment]
For ML rules, it's often required to register proper partitions, too, and the presentation reconciler should subsequently work for specific partition types. See IDocumentPartitioner and its various friends.

**Comment by laeubi** [comment]
@szarnekow can you maybe help in fixing the PDE implementation? I want to to make m2e to support more advanced features for its contribution to the target editor but currently it does not work because PDE "hack" :-\
SO maybe it is not a bug in [PresentationReconciler](https://github.com/eclipse-platform/eclipse.platform.ui/issues/1330#top) but PDE is not setting up the document correctly?!?
